core/src/main/java/com/dtolabs/rundeck/core/common/PluginControlServiceImpl.java

```diff
@@ -1,27 +1,31 @@
 package com.dtolabs.rundeck.core.common;
 
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.HashSet;
-import java.util.List;
-import java.util.Set;
+import com.dtolabs.rundeck.core.plugins.PluginRegistry;
+
+import java.util.*;
 import java.util.function.Predicate;
 
 public class PluginControlServiceImpl implements PluginControlService {
 
     public static final String DISABLED_PLUGINS = "disabled.plugins";
     private HashSet<String> disabledPlugins;
     private final IFramework framework;
+    private final PluginRegistry pluginRegistry;
     private final String project;
 
-    private PluginControlServiceImpl(IFramework framework, String project) {
+    private PluginControlServiceImpl(IFramework framework, PluginRegistry pluginRegistry, String project) {
         this.framework = framework;
+        this.pluginRegistry = pluginRegistry;
         this.project = project;
     }
 
-
+    @Deprecated
     public static PluginControlService forProject(IFramework framework, String project) {
-        return new PluginControlServiceImpl(framework, project);
+        return new PluginControlServiceImpl(framework, null, project);
+    }
+
+    public static PluginControlService forProject(IFramework framework, PluginRegistry pluginRegistry, String project) {
+        return new PluginControlServiceImpl(framework, pluginRegistry, project);
```

Comment:
If we decide we like this approach, there is still a small TODO to update the `public Predicate<String> enabledPredicateForService(final String serviceName)` method so that it accounts for the `pluginRegistry` when available.